src/creature_states_hero.c

```diff
@@ -975,7 +975,12 @@ short good_doing_nothing(struct Thing *creatng)
             cctrl->hero.byte_8C = 1;
         }
         nturns = game.play_gameturn - cctrl->hero.look_for_enemy_dungeon_turn;
-        if (nturns > 64)
+        if ((nturns > 64) && (cctrl->unknown.byte_8B == 1))
```

Comment:
I guess the unknown block of this union can be dropped and replaced with the hero block, also name byte_8B map_changed_for_nagivation or something instead of leaving it as pseudo code

Reply:
As discussed on discord, the unions are more complex then they seem at first look, possibly with bugs introduced at some places, and it is not clear to me what to change/fix.